src/Doctrine/Orm/Filter/AbstractFilter.php

```diff
@@ -53,29 +58,43 @@ public function apply(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $q
      */
     abstract protected function filterProperty(string $property, $value, QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass, ?Operation $operation = null, array $context = []): void;
 
-    protected function getManagerRegistry(): ManagerRegistry
+    public function hasManagerRegistry(): bool
+    {
+        return $this->managerRegistry instanceof ManagerRegistry;
+    }
+
+    public function getManagerRegistry(): ManagerRegistry
     {
+        if (!$this->hasManagerRegistry()) {
+            throw new \RuntimeException('ManagerRegistry must be initialized before accessing it.');
```

Comment:
```suggestion
            throw new RuntimeException('ManagerRegistry must be initialized before accessing it.');
```

use ApiPlatform\Metadata\RuntimeException